### PR TITLE
Web Proxy Support

### PR DESCRIPTION
**Pull request by DarcyRaynerDD**
### What does this PR do?

Adds support for HTTP_PROXY and HTTPS_PROXY env vars. These are used for setting the web proxy to send requests through. This PR also fixes bugs with sending metrics and traces while using web proxy.

### Motivation

Fixing web proxy support.

### Testing Guidelines

I manually tested with both wtih a VPC and web proxy setup, and without.

### Types of changes

- [X] Bug fix
- [X] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [X] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [X] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [ ] This PR passes the unit tests 
- [ ] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: aws/logs_monitoring/template.yaml [hunk 1]
```diff
@@ -154,6 +154,10 @@ Parameters:
       - true
       - false
     Description: Set to true to deploy the Forwarder to a VPC and send logs, metrics, and traces via a proxy. When set to true, must also set VPCSecurityGroupIds and VPCSubnetIds.
+  DdHttpProxyURL:
+    Type: String
+    Default: ""
+    Description: "For web proxy support when using a VPC, sets the HTTP_PROXY and HTTPS_PROXY values. Don't use this in combination with AWS Private Link. When set, must also set VPCSecurityGroupIds and VPCSubnetIds."
```
Comment (tianchu) [inline aws/logs_monitoring/template.yaml]: Can you add one example? so user have higher confidence knowing what they put in is in the expected format.

File: aws/logs_monitoring/template.yaml [hunk 2]
```diff
@@ -282,10 +286,17 @@ Conditions:
...
+  SetDdHttpProxyURL:
+    Fn::Not:
+      - Fn::Equals:
+          - Ref: DdHttpProxyURL
+          - ""
   UseVPC:
     Fn::Or:
       - Condition: SetDdUsePrivateLink
       - Condition: SetDdUseVPC
+      - Condition: SetDdHttpProxyURL
```
Comment (tianchu) [inline aws/logs_monitoring/template.yaml]: As discussed, i don't think `SetDdHttpProxyURL` should force `UseVPC`. I expect the user to turn on `DdUseVPC` if it's indeed in a VPC.

File: aws/logs_monitoring/template.yaml [hunk 3]
```diff
@@ -154,6 +154,10 @@ Parameters:
       - true
       - false
     Description: Set to true to deploy the Forwarder to a VPC and send logs, metrics, and traces via a proxy. When set to true, must also set VPCSecurityGroupIds and VPCSubnetIds.
+  DdHttpProxyURL:
```
Comment (tianchu) [inline aws/logs_monitoring/template.yaml:157]: Can you add this parameter to the README? Both the parameter reference table and the section discussing proxy options?

**Review by tianchu — APPROVED** [review]

File: aws/logs_monitoring/README.md [hunk 4]
```diff
@@ -212,18 +217,25 @@ You can run the Forwarder in a VPC by using AWS PrivateLink to connect to Datado
...
+2. Update your proxy with following configurations ([HAProxy](https://github.com/DataDog/datadog-serverless-functions/blob/master/aws/logs_monitoring/proxy_conf/haproxy.txt) or [Nginx](https://github.com/DataDog/datadog-serverless-functions/blob/master/aws/logs_monitoring/proxy_conf/nginx.txt)). If you are using another proxy, or Web Proxy, whitelist the datadog domain eg. .datadoghq.com .
 3. When installing the Forwarder with the CloudFormation template, set `DdUseVPC`, `VPCSecurityGroupIds` and `VPCSubnetIds`.
 4. Ensure the `DdFetchLambdaTags` option is disabled, because AWS VPC does not yet offer an endpoint for the Resource Groups Tagging API.
-5. Set `DdApiUrl` to `http://<proxy_host>:3834` or `https://<proxy_host>:3834`.
-6. Set `DdTraceIntakeUrl` to `http://<proxy_host>:3835` or `https://<proxy_host>:3835`.
-7. Set `DdUrl` to `<proxy_host>` and `DdPort` to `3837`.
-8. Set `DdNoSsl` to `true` if connecting to the proxy using `http`.
-9. Set `DdSkipSslValidation` to `true` if connecting to the proxy using `https` with a sef-signed certificate.
+5. If using a HAProxy of Nginx
```
Comment (nhinsch) [inline aws/logs_monitoring/README.md]: If using a HAProxy of Nginx -> If using HAProxy or Nginx

File: aws/logs_monitoring/README.md [hunk 5]
```diff
@@ -212,18 +217,25 @@ You can run the Forwarder in a VPC by using AWS PrivateLink to connect to Datado
...
+
+- Set `DdApiUrl` to `http://<proxy_host>:3834` or `https://<proxy_host>:3834`.
+- Set `DdTraceIntakeUrl` to `http://<proxy_host>:3835` or `https://<proxy_host>:3835`.
+- Set `DdUrl` to `<proxy_host>` and `DdPort` to `3837`.
+
+6. Otheriwse, if using Web Proxy
```
Comment (nhinsch) [inline aws/logs_monitoring/README.md]: I would combine steps 5 and 6. They are both basically the same thing, just for different proxies.

Also, Otheriwse -> Otherwise

File: aws/logs_monitoring/README.md [hunk 6]
```diff
@@ -344,6 +352,9 @@ To test different patterns against your logs, turn on [debug logs](#troubleshoot
 `DdUsePrivateLink`
 : Set to true to enable sending logs and metrics via AWS PrivateLink. See https://dtdg.co/private-link.
 
+`DdHttpProxyURL`
+: For web proxy support, sets the HTTP_PROXY and HTTPS_PROXY values. Don't use this in combination with AWS Private Link. Make sure to also set DdSkipSslValidation to true, and
+
```
Comment (nhinsch) [inline aws/logs_monitoring/README.md:359]: I think this sentence is incomplete (ends with "and")
That said, I think I would rephrase this. What are the `HTTP_PROXY` and `HTTPS_PROXY` values? Why do they matter? It might be better to back up a bit and explain a bit what this setting is for at a higher level.

Comment (DarcyRaynerDD) [reply aws/logs_monitoring/README.md:359]: I added a little bit of information, but it's kind of a big topic to cover. It should be clear to someone who has set up web proxy, what those variables are for. I think with this, plus the steps in the proxy in section, it should be enough to cover how to use it.
